modules/react-native-status/ios/RCTStatus/RCTStatus.m

```diff
@@ -95,9 +118,21 @@ -(void)setBridge:(RCTBridge *)newBridge
     NSLog(@"CallJail() method called");
 #endif
     dispatch_async( dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        char * result = Call((char *) [chatId UTF8String], (char *) [path UTF8String], (char *) [params UTF8String]);
+        
+        NSString *stringResult;
+        if([Status JSCEnabled]){
+            if(_jail == nil) {
+                _jail = [Jail new];
+            }
+            NSDictionary *result = [_jail call:chatId path:path params:params];
```

Comment:
How come the return type is NSDictionary as opposed to char*? Doesn't *result in both places serve the same purpose?

Reply:
i would prefer to not use char* at all but that's the way how we communicate with c layer. For code related to JSC we don't need this, so it would be either NSString or NSDictionary. I don't understand what is the problem here

Reply:
Ok no problem, just seemed like change in interface for same functionality. Makes sense re C layer

Reply:
Yes i agree about interface, as you see i opt for three `if` statement instead of some interface because i'm not 100% sure how to rewrite it better. So i don't pretend that this code is perfect or even good (especially obj-c), thus it would be great if some proper refactoring will be contributed later.